### PR TITLE
New package: bastet-0.43.2

### PR DESCRIPTION
**Pull request by awest03**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - i686
  - armv6l

### CODE REVIEW

File: srcpkgs/bastet/template [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Template file for 'bastet'
+pkgname=bastet
+version=0.43.2
+revision=1
+archs="~*-musl"
+hostmakedepends="gcc"
```
Comment (Johnnynator) [inline srcpkgs/bastet/template:6]: gcc is never needed in `hostmakedepend`

File: srcpkgs/bastet/template [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+maintainer="Alaisdair West <alaisdair.west2@outlook.com>"
+license="GPL-3.0-only"
+homepage="http://fph.altervista.org/prog/bastet.html"
+changelog="https://raw.githubusercontent.com/fph/bastet/master/NEWS"
+distfiles="https://github.com/fph/${pkgname}/archive/refs/tags/${version}.tar.gz"
+checksum=f219510afc1d83e4651fbffd5921b1e0b926d5311da4f8fa7df103dc7f2c403f
+nocheckperms=yes # For High Score file
+
+do_build() {
+	make
```
Comment (Johnnynator) [inline srcpkgs/bastet/template:19]: You probably want to use `build_style=gnu-makefile`

File: srcpkgs/bastet/template [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Template file for 'bastet'
+pkgname=bastet
+version=0.43.2
+revision=1
+archs="~*-musl"
```
Comment (Johnnynator) [inline srcpkgs/bastet/template:5]: Can you privide the reason musl is disabled?

**Comment by awest03** [comment]
Sorry I've messed up the commit history so I have marked it was WIP while I fix it.

File: srcpkgs/bastet/template (hunk 3 above)
Comment (awest03) [reply srcpkgs/bastet/template:5]: Building with musl encountered some errors, I'll try again and see if I can work out a patch for it

Comment (awest03) [reply srcpkgs/bastet/template]: I have re-enabled it now, turns out I had something set up wrong and it builds with musl now, sorry

File: srcpkgs/bastet/template (hunk 2 above)
Comment (awest03) [reply srcpkgs/bastet/template:19]: Unfortunately the makefile provided by bastet does not seem to work with that

File: srcpkgs/bastet/patches/new-boost.patch [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+! #include <boost/unordered_set.hpp>
+  #include <boost/functional/hash.hpp>
+*************** namespace Bastet{
+*** 77,79 ****
+    private:
+!     std::tr1::unordered_set<Vertex> _visited;
+      //std::set<Vertex> _visited; ^^ the above is more efficient, we need to do many inserts
+--- 76,78 ----
+    private:
+!     boost::unordered_set<Vertex> _visited;
```
Comment (sgn) [inline srcpkgs/bastet/patches/new-boost.patch:21]: Why not `std::unordered_set`?

Comment (awest03) [reply srcpkgs/bastet/patches/new-boost.patch:21]: `boost::unordered_set` is used in the repository (however unreleased) and was used to patch for FreeBSD, so I assumed that this was the appropriate solution.

File: srcpkgs/bastet/template (hunk 3 above)
Comment (awest03) [reply srcpkgs/bastet/template]: I've managed to patch it to work with musl and built for x86_64-musl

File: srcpkgs/bastet/patches/new-boost.patch (hunk 4 above)
Comment (sgn) [reply srcpkgs/bastet/patches/new-boost.patch:21]: `boost::unorder_set` needs to be linked with libboost_container, I think

**Pull request closed without merging**